projects/packages/jetpack-mu-wpcom/src/features/calypso-locale-sync/calypso-locale-sync.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Sync Calypso locale when wp-admin locale is updated via /wp-admin/profile.php.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+if ( function_exists( 'wpcom_is_nav_redesign_enabled' ) && wpcom_is_nav_redesign_enabled() ) {
+	add_filter(
+		'insert_user_meta',
```

Comment:
Nitpick (non-blocking): Would it be easier to use the `switch_locale` hook instead? https://developer.wordpress.org/reference/hooks/switch_locale/

Reply:
I could be wrong about this, but I think `switch_locale` can be trigger on "temporary" locale switches, like viewing a page in locale other than what's in your profile settings. If that's true, we may want to stick with `insert_user_meta` since it's triggered when a user deliberately changes their locale setting.